internal/envoy/cluster.go

```diff
@@ -51,6 +60,22 @@ func Cluster(c *dag.Cluster) *v2.Cluster {
 	}
 }
 
+func upstreamValidationCACert(upstream *dag.HTTPService) []byte {
+	if upstream.UpstreamValidation != nil {
```

Comment:
Please change these conditions so that the success case proceeds down the page. The guard clause should be 
```go
if upstreamValidation == nil {
   // right, it's nil, what next
  return nil
}
return upstream.UpstreamValidation.CACertificate.Object.Data[CACertificateKey]